### PR TITLE
Fix path in error messages on sketch loading

### LINKED ISSUE #1797: Incorrect error `details` when the `LoadSketchRequest` fails with `NOT_FOUND` error
### Describe the problem

If I have such a gRPC `LoadSketchRequest`:

```json
{
    "instance": {
        "id": 1
    },
    "sketchPath": "/Users/a.kitta/Documents/Arduino/alma"
}
```

And the folder is missing from the file system:
```
 % cd ~/Documents/Arduino/alma
cd: no such file or directory: /Users/a.kitta/Documents/Arduino/alma
```

Then the request correctly fails with a [`NOT_FOUND`](https://grpc.github.io/grpc/core/md_doc_statuscodes.html) status code but the error `details` is incorrect:
```json
{
    "code": 5,
    "details": "Can't open sketch: no valid sketch found in /Users/a.kitta/Documents/Arduino: missing /Users/a.kitta/Documents/Arduino/Arduino.ino",
    "metadata": {
        "content-type": [
            "application/grpc"
        ]
    }
}
```

The error message should be about `Arduino/alma/alma.ino` and not `Arduino/Arduino.ino`. The `directories.user` location exists on my disk:
```
% ls -al ~/Documents/Arduino 
total 0
drwxr-xr-x  2 a.kitta  staff   64 Jul  8 16:50 .
drwx------+ 5 a.kitta  staff  160 Jul  6 15:46 ..
```

### To reproduce

 - Get the IDE2 build from here: https://github.com/arduino/arduino-ide/pull/1152
 - Create a new sketch and save it,
 - Quit IDE2,
 - Delete the previously deleted sketch and start IDE2,
 - You will see the incorrect error message

Screen Shot 2022-07-08 at 17 05 58

### Expected behavior

It should complain about `~/Documents/Arduino/alma` and not `~/Documents/Arduino/`

### Arduino CLI version

`0.24.0`

### Operating system

macOS

### Operating system version

12.3.1

### Additional context

_No response_

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-cli/issues?q=)
- [X] I verified the problem still occurs when using the [nightly build](https://arduino.github.io/arduino-cli/dev/installation/#nightly-builds)
- [X] My report contains all necessary details

### PR DESCRIPTION
**Pull request by cmaglie**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

**What kind of change does this PR introduce?**
* Fix #1797
* Removes a compatibility layer for the Arduino IDE 1.x, see https://github.com/arduino/arduino-cli/pull/1187 for details. The proper fix must be made in the Arduino IDE 1.x, not as an exception in the CLI.

**What is the current behavior?**

```
$ arduino-cli version
arduino-cli.exe  Version: git-snapshot Commit: 6ac5f7a3 Date: 2022-07-09T19:16:18Z

$ mkdir /tmp/Arduino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma
Error opening sketch: no valid sketch found in C:\Users\per\AppData\Local\Temp\Arduino: missing C:\Users\per\AppData\Local\Temp\Arduino\Arduino.ino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/alma.ino
Error opening sketch: reading files: open C:\Users\per\AppData\Local\Temp\Arduino\alma: The system cannot find the file specified.

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/nonexistent
Error opening sketch: reading files: open C:\Users\per\AppData\Local\Temp\Arduino\alma: The system cannot find the file specified.
```

**What is the new behavior?**
```
$ arduino-cli version
arduino-cli.exe  Version: git-snapshot Commit: 6ac5f7a3 Date: 2022-07-09T19:16:18Z

$ mkdir /tmp/Arduino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma
Error opening sketch: main file missing from sketch: /tmp/Arduino/alma/alma.ino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/alma.ino
Error opening sketch: main file missing from sketch: /tmp/Arduino/alma/alma.ino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/nonexistent
Error opening sketch: main file missing from sketch: /tmp/Arduino/alma/nonexistent/nonexistent.ino
```

**Does this PR introduce a breaking change, and is
[titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**
No

### CODE REVIEW

**Comment by kittaakos** [comment]
> main file missing from sketch: /tmp/Arduino/alma/alma.ino

I am not sure about it. The problem is not the missing main file from the sketch, but the sketch does not exist.

**Comment by cmaglie** [comment]
Ok, it should be better now:

```
$ mkdir /tmp/Arduino
$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma
Error opening sketch: no such file or directory: /tmp/Arduino/alma

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/alma.ino
Error opening sketch: no such file or directory: /tmp/Arduino/alma/alma.ino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/nonexistent
Error opening sketch: no such file or directory: /tmp/Arduino/alma/nonexistent

$ mkdir /tmp/Arduino/alma

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma
Error opening sketch: main file missing from sketch: /tmp/Arduino/alma/alma.ino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/alma.ino
Error opening sketch: no such file or directory: /tmp/Arduino/alma/alma.ino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/nonexistent
Error opening sketch: no such file or directory: /tmp/Arduino/alma/nonexistent

$ touch /tmp/Arduino/alma/asd.ino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma
Error opening sketch: main file missing from sketch: /tmp/Arduino/alma/alma.ino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/asd.ino 
Error opening sketch: main file missing from sketch: /tmp/Arduino/alma/alma.ino

$ arduino-cli compile -b arduino:avr:uno /tmp/Arduino/alma/nonexistent
Error opening sketch: no such file or directory: /tmp/Arduino/alma/nonexistent
```

File: arduino/sketch/sketch.go [hunk 1]
```diff
@@ -64,6 +64,11 @@ func New(path *paths.Path) (*Sketch, error) {
 	}
 
 	path = path.Canonical()
+	if exist, err := path.ExistCheck(); err != nil {
+		return nil, fmt.Errorf("%s: %s", tr("sketch path is not valid"), err)
```
Comment (umbynos) [inline arduino/sketch/sketch.go:68]: maybe it would be cool to have a test also for this?

Comment (cmaglie) [reply arduino/sketch/sketch.go:68]: It's a bit hard test this branch, under the hood `ExistCheck` does a `Stat` call, but I don't know how to make it fail... if you have any idea... :-)

Comment (umbynos) [reply arduino/sketch/sketch.go:68]: Mhh you are right.. It seems difficult

**Review by umbynos — APPROVED** [review]

**Pull request merged**